### PR TITLE
Migration of palette elements

### PR DESCRIPTION
**Pull request by mike-spa**
Resolves: #17581 

@cbjeukendrup @igorkorsukov I've attempted a very basic solution here which seems to work nicely. Let me know if it's ok or we need something different.

### CODE REVIEW

File: src/palette/internal/palettecell.cpp
Comment (igorkorsukov) [inline src/palette/internal/palettecell.cpp:233]: I think it's better to add a separate class with name like "PaletteCompat" and add this function to it as static with input arg and output (new item);
In the future, we may add other fefunctions to this class and we will see what kind of compatibility we support.

Comment (mike-spa) [reply src/palette/internal/palettecell.cpp]: @igorkorsukov done! 👍

**Review by igorkorsukov — APPROVED** [review]

**Comment by mike-spa** [comment]
@igorkorsukov the builds and tests are now all passing, feel free to merge this, if it's ok

**Review by cbjeukendrup — APPROVED** [review]
FWIW, looks good to me too!

File: src/engraving/rw/compat/compatutils.h
Comment (igorkorsukov) [reply src/engraving/rw/compat/compatutils.h:49]: Yes, I think it's better to be UPPER_CASE.
But I think this can be done in some other PR, if we do not need to modify this

**Review by oktophonie — APPROVED** [review]

**Comment by DmitryArefiev** [comment]
@mike-spa So the way to check is run 4.0.2 build reset text palette (should be **E**xpression), then run PR's build - it should should show **e**xpression without Text palette reset, correct?

**Comment by DmitryArefiev** [comment]
I found how to check it. Run 4.0.2 build (dev), reset ornament palette, run PR's build, add an ornament to score, check available options in Properties: they should have an updated style (without Reset ornament palette)

@mike-spa I see the branch has a conflicts, should be resolved before merge?

**Comment by mike-spa** [comment]
> @mike-spa So the way to check is run **4.0.2 build** reset text palette (should be '**E**xpression'), then run **PR's build** - it should show '**e**xpression' without Text palette reset, correct?

@DmitryArefiev yes, sorry I missed this comment. That's exactly the way to test it, and also the other one you did with expressions. Thank you for testing this. As for the conflict, I'll rebase this and fix it asap

**Comment by mike-spa** [comment]
done!

**Comment by DmitryArefiev** [comment]
Cool! Thanks!

Tested #17581 on Win10, Mac13 - **PASS**

Merged

**Pull request merged**
